src/EnergyPlus/DataHeatBalSurface.cc

```diff
@@ -93,6 +93,21 @@ namespace DataHeatBalSurface {
     Real64 MaxSurfaceTempLimitBeforeFatal(500.0); // 2.5 times MaxSurfaceTempLimit
     Array1D<Real64> CTFConstInPart;               // Constant Inside Portion of the CTF calculation
     Array1D<Real64> CTFConstOutPart;              // Constant Outside Portion of the CTF calculation
+    // This group of arrays (soon to be vectors) added to facilitate vectorizable loops in CalcHeatBalanceInsideSurf2CTFOnly
+    Array1D<Real64> CTFCross0;                    // Construct.CTFCross(0)
+    Array1D<Real64> CTFInside0;                   // Construct.CTFInside(0)
+    Array1D<Real64> CTFSourceIn0;                 // Construct.CTFSourceIn(0)
+    Array1D<Real64> TH11Surf;                     // TH(1,1,SurfNum)
+    Array1D<Real64> QsrcHistSurf1;                // QsrcHist(SurfNum, 1)
+    Array1D_int IsAdiabatic;                      // 0 not adiabatic, 1 is adiabatic
+    Array1D_int IsNotAdiabatic;                   // 1 not adiabatic, 0 is adiabatic
+    Array1D_int IsSource;                         // 0 no internal source/sink, 1 has internal source/sing
+    Array1D_int IsNotSource;                      // 1 no internal source/sink, 0 has internal source/sing
+    Array1D_int IsPoolSurf;                       // 0 not pool, 1 is pool
+    Array1D_int IsNotPoolSurf;                    // 1 not pool, 0 is pool
+    Array1D<Real64> TempTermSurf;                 // TempTerm for heatbalance equation
+    Array1D<Real64> TempDivSurf;                  // Divisor for heatbalance equation
+    // end group added to support CalcHeatBalanceInsideSurf2CTFOnly
```

Comment:
These 1d arrays were added to facilitate auto-vectorization in the new function `CalcHeatBalanceInsideSurf2CTFOnly`. The integer 1/0 terms are used to include/exclude different terms as needed for various surface types in general surface heat balance equations. As the heat balance data is re-structured into simple vectors, these will likely be modified. @xuanluo113 @Myoldmopar

Reply:
Is there a reason you didn't go with arrays of bools rather than integers?

Reply:
I chose integers because these are multiplied by reals in the heat balance equations. The zero terms become zero and drop out, but it's all just one long equation so the compiler can auto-vectorize. e.g.
```
TempDivSurf(surfNum) =
                    1.0 / (CTFInside0(surfNum) - IsAdiabatic(surfNum) * CTFCross0(surfNum) + IsPoolSurf(surfNum) * PoolHeatTransCoefs(surfNum) +
                           IsNotPoolSurf(surfNum) * HConvIn(surfNum) + iterDampConstant);
```
This does introduce some extra calculations. Perhaps there's a better way. And I started with Array1D a while back before we decided to move away from those; just didn't circle back to change them.